sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -0,0 +1,437 @@
+        ///
+        protected virtual Task OnInitializingPartitionAsync(TPartition partition,
+                                                            CancellationToken cancellationToken) => Task.CompletedTask;
+
+        /// <summary>
+        ///   Performs the tasks needed when processing for a partition is being stopped.  This commonly occurs when the partition
+        ///   is claimed by another event processor instance or when the current event processor instance is shutting down.
+        /// </summary>
+        ///
+        /// <param name="partition">The context of the partition for which processing is being stopped.</param>
+        /// <param name="reason">The reason that processing is being stopped for the partition.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the processing.  This is not expected to signal under normal circumstances and will only occur if the processor encounters an unrecoverable error.</param>
+        ///
+        protected virtual Task OnPartitionProcessingStoppedAsync(TPartition partition,
+                                                                 ProcessingStoppedReason reason,
+                                                                 CancellationToken cancellationToken) => Task.CompletedTask;
+
+        /// <summary>
+        ///   A set of information about the last enqueued event of a partition, as observed by the associated EventHubs client
+        ///   associated with this context as events are received from the Event Hubs service.  This is only available if the consumer was
+        ///   created with <see cref="ReadEventOptions.TrackLastEnqueuedEventProperties" /> set.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition to read the properties from.</param>
+        ///
+        /// <returns>The set of properties for the last event that was enqueued to the partition.</returns>
+        ///
+        /// <remarks>
+        ///   When information about the partition's last enqueued event is being tracked, each event received from the Event Hubs
+        ///   service will carry metadata about the partition that it otherwise would not. This results in a small amount of
+        ///   additional network bandwidth consumption that is generally a favorable trade-off when considered
+        ///   against periodically making requests for partition properties using an Event Hub client.
+        /// </remarks>
+        ///
+        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="EventProcessorOptions.TrackLastEnqueuedEventProperties" /> set or when the processor is not running.</exception>
```

Comment:
Is there any implementation detail that makes it hard to retrieve this info once the processor has stopped? Why would we want to make users unable to call this method in this scenario?

Reply:
It comes directly from the transport consumer that we're disposing when we stop.  We could do a local cache when we stop and use that, but unless we have feedback calling for it, I'd prefer not to remove that constraint.